demo/admin/src/dam/ImportFromUnsplash.tsx

```diff
@@ -0,0 +1,79 @@
+import { CancelButton, messages, SaveButton } from "@comet/admin";
+import { useCurrentDamFolder, useDamAcceptedMimeTypes, useDamFileUpload } from "@comet/cms-admin";
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { getRandomUnsplashImage, UnsplashImage } from "../dam/getRandomUnsplashImage";
+import UnsplashIcon from "./UnsplashIcon";
+
+export const ImportFromUnsplash: React.FC = () => {
+    const { allAcceptedMimeTypes } = useDamAcceptedMimeTypes();
+    const { folderId } = useCurrentDamFolder();
+    const [isOpen, setIsOpen] = React.useState(false);
+    const [unsplashImage, setUnsplashImage] = React.useState<UnsplashImage>();
+
+    const { uploadFiles } = useDamFileUpload({
+        acceptedMimetypes: allAcceptedMimeTypes,
+    });
+
+    const handleOpenDialog = async () => {
+        const image = await getRandomUnsplashImage();
+        setUnsplashImage(image);
+        setIsOpen(true);
+    };
+
+    const handleCloseDialog = () => {
+        setIsOpen(false);
+    };
+
+    const handleSave = () => {
+        if (unsplashImage === undefined) return;
+        uploadFiles(
```

Comment:
(minor, demo only): you should await this before closing the dialog

Reply:
Fixed: 1a1d22cb98f760dc4a474db4093cb2559eabeebf